libraries/core/src/main/kotlin/com/bumble/appyx/core/composable/Children.kt

```diff
@@ -50,18 +58,42 @@ inline fun <reified NavTarget : Any, State> ParentNode<NavTarget>.Children(
             )
         }
     }
-    Box(modifier = modifier
-        .onSizeChanged {
-            transitionBounds = it
+    if (withSharedElementTransition) {
```

Comment:
I assume `withSharedElementTransition` won't change dynamically that often but how the API is set up, it _could_. Should we wrap the repeated code (i.e. `CompositionLocalProvider(...) { block(...) }`) with the `moveableContentOf` API to 1) avoid repeating this code and 2) save state if `withSharedElementTransition` were to change?

Reply:
I can extract repeatable parts to a separate composable but it doesn't make sense to you `moveableContentOf` the layout either supports shared element transition or not. I can't come up with a case when we need dynamically change it. WDYT?

Reply:
The use case could be that an app allows the user to enable/disable animations on the fly. It's definitely an edge case and something that wouldn't happen very often. But given this is an open source Architecture library and this affects the public API, I guess people could use it for _anything_. If I change `withSharedElementTransition` on the fly, I wouldn't want to lose state.